pkg/virt-controller/watch/vm.go

```diff
@@ -2717,21 +2717,22 @@ func (c *VMController) handleDynamicInterfaceRequests(vm *virtv1.VirtualMachine,
 	for i := range vm.Status.InterfaceRequests {
 		vmTemplateCopy = controller.ApplyNetworkInterfaceRequestOnVMISpec(
 			vmTemplateCopy, &vm.Status.InterfaceRequests[i])
+	}
+	vm.Spec.Template.Spec = *vmTemplateCopy
 
-		if vmi == nil || vmi.DeletionTimestamp != nil {
-			continue
-		}
-
+	if vmi != nil && vmi.DeletionTimestamp == nil {
 		vmiSpecCopy := vmi.Spec.DeepCopy()
-		vmiSpecCopy = controller.ApplyNetworkInterfaceRequestOnVMISpec(
-			vmiSpecCopy, &vm.Status.InterfaceRequests[i])
+		for i := range vm.Status.InterfaceRequests {
+			vmiSpecCopy = controller.ApplyNetworkInterfaceRequestOnVMISpec(
+				vmiSpecCopy, &vm.Status.InterfaceRequests[i])
+		}
+		vmi.Spec = *vmiSpecCopy
```

Comment:
This seems wrong, I do not understand how it works at all now.

Reply:
This is an intermediate step, please see the following commits

Reply:
> This seems wrong, I do not understand how it works at all now.
Can you please be more specific?

Basically what it does is very similar to the VM template sepc.
It copies the VMI spec, mutates it according to the interface requests, and finally patch the VMI object based on the copied mutated spec.

Reply:
The VM template spec is being updated in this method, it is not used to patch the VM.

The original `vmi` should not be updated, if it is (like done here) it will make the patching of the VMI function do nothing.

Even if this is an intermediate change, which later the whole patching is gone, it is per what I suspect incorrect. Rebasing over this commit may be then harder or just simply buggy.

Reply:
IMO the review should be focused on this change, which is the basis of other changes done in another PR.
I would recommend to review this though https://github.com/kubevirt/kubevirt/pull/9975 to avoid such issues.

Reply:
You right, the missing part is passing a copy for the VMI object, it probably been removed on the last rebase and works because it removed later on..

Reply:
Fixed

Reply:
This PR been rebased, this change already exist on main branch so this commit is removed.